src/geometry/CTiglIntersectionCalculation.cpp

```diff
@@ -136,6 +162,67 @@ CTiglIntersectionCalculation::CTiglIntersectionCalculation(CTiglShapeCache* cach
     computeIntersection(cache, hash1, hash2, shape, plane);
 }
 
+// Computes the intersection of a shape with a plane segment,
+// that is defined by two points p1 and p2 and a vector w in
+// the following way: p(u,v) = p1 (1-u) + p2u + wv, with
+// u in [0,1]. Ideally w should be perpendicular to (p1 - p2)
+// cache variable can be NULL
+CTiglIntersectionCalculation::CTiglIntersectionCalculation(CTiglShapeCache* cache,
+                                                           const std::string& shapeID,
+                                                           TopoDS_Shape shape,
+                                                           gp_Pnt point1,
+                                                           gp_Pnt point2,
+                                                           gp_Dir normal,
+                                                           bool forceOrthogonal)
+    : tolerance(1.0e-7)
+{
+
+    // create hash
+    size_t hash = boost::hash<std::string>()(shapeID);
+    size_t hashP1P2 = HashableProjection(point1,normal).HashValue() ^ HashableProjection(point2,normal).HashValue();
```

Comment:
I had a look on my mac to inspect the hash collisions. I could not really understand, why the xor in this case results in the same hash, but this is reality. I thought and read about it a bit, and xor seems not to be a good solutions, since ```a xor a = 0```and also ```b xor b = 0```. This means, that there might be many collisions for symmetric arguments.
To fix this, we could just use ```+```, which also is a commutative way to combine the hashes. Please replace all "^" with "+" in this file. The test cases seem to work better with "+".